airflow/models/taskinstance.py

```diff
@@ -1376,17 +1375,20 @@ def xcom_pull(
         if dag_id is None:
             dag_id = self.dag_id
 
-        pull_fn = functools.partial(
-            XCom.get_one,
+        result = XCom.compose_query(
             execution_date=self.execution_date,
             key=key,
             dag_id=dag_id,
-            include_prior_dates=include_prior_dates)
+            task_id=task_ids,
+            include_prior_dates=include_prior_dates
+        )
 
         if is_container(task_ids):
-            return tuple(pull_fn(task_id=t) for t in task_ids)
+            return [XCom.deserialize_value(xcom) for xcom in result.all()]
```

Comment:
With `@reconstructor` applied to Xcom.init_on_load do we actually need to call this here? Isn't this handled for us already?

Reply:
Good point Ash, I've missed the `@reconstructor` and this caused tests to fail. I've added the `.with_entities(XCom.value)` to the query, and therefore we bypass the reconstructor. I must admit that I'm not a big fan of the `@reconstructor` in general (Explicit is better than implicit). The `.with_entities` will only fetch the `value` object, therefore we only select this field from the database.

Reply:
Ah I missed the `.with_entities(XCom.value)` in the big hunk of a diff. I think that should be applied here, not inside `compose_query()`

Reply:
Yes, good point. I'll move that out and add some comments.